src/artifact_providers/__tests__/github.test.ts

```diff
@@ -0,0 +1,211 @@
+jest.mock('../../utils/githubApi.ts');
+import { getGithubClient } from '../../utils/githubApi';
+import { GithubArtifactProvider, ArtifactItem } from '../github';
+
+class TestGithubArtifactProvider extends GithubArtifactProvider {
+  public testGetRevisionArtifact(revision: string): Promise<ArtifactItem> {
+    return this.getRevisionArtifact(revision);
+  }
+}
```

Comment:
What's the value here vs. using `GithubArtifactProvider` directly?

Reply:
It is not a value, it is a necessity if we want to keep the `getRevisionArtifact` method private on the class. I switched it to `protected` so it is exposed to child classes but not on the instances.